.github/copilot-instructions.md

```diff
@@ -0,0 +1,252 @@
+# Devito Copilot Instructions
+
+## Repository Overview
+
+Devito is a Python package for implementing optimized stencil computation (finite differences, image processing, machine learning) from high-level symbolic problem definitions. It builds on SymPy and employs automated code generation and just-in-time compilation to execute optimized computational kernels on CPUs, GPUs, and clusters.
+
+**Key Facts:**
+- Production-stable package (Development Status :: 5 - Production/Stable)
+- Python 3.10-3.13 support with comprehensive CI testing
+- ~290MB repository with extensive test coverage and benchmarking
+- Primary languages: Python (core), C/C++ (generated code)
+- Main dependencies: NumPy, SymPy, cgen, codepy, psutil
+- Supports multiple compilation backends: C, OpenMP, OpenACC, MPI
+- Active development with 600+ stars, 238+ forks, and regular releases
+
+## Build and Development Environment
+
+### Prerequisites and Installation
+
+**System Dependencies (Ubuntu/Debian):**
+```bash
+sudo apt update
+sudo apt install -y python3-numpy python3-sympy python3-psutil python3-pip
+sudo apt install -y gcc g++ flake8
+```
+
+**Development Installation (preferred method):**
+```bash
+# Install with all dependencies - requires network access
+pip install -e .[tests,extras] --break-system-packages
+
+# Alternative: conda environment (recommended for full development)
+conda env create -f environment-dev.yml
+conda activate devito
```

Comment:
This is overall no good, it installs the deps before the venv. 

- Use python venv instead (and install python3-venv above)
- No need for --break-system-packages in a ven since it's not system anymore.